modules/dnn/src/op_inf_engine.cpp

```diff
@@ -652,20 +656,24 @@ InferenceEngine::Core& getCore(const std::string& id)
 #endif
 
 #if !defined(OPENCV_DNN_IE_VPU_TYPE_DEFAULT)
-static bool detectMyriadX_()
+static bool detectMyriadX_(std::string device)
 {
     AutoLock lock(getInitializationMutex());
 #if INF_ENGINE_VER_MAJOR_GE(INF_ENGINE_RELEASE_2019R3)
     // Lightweight detection
-    InferenceEngine::Core& ie = getCore("MYRIAD");
+    InferenceEngine::Core& ie = getCore(device);
     const std::vector<std::string> devices = ie.GetAvailableDevices();
     for (std::vector<std::string>::const_iterator i = devices.begin(); i != devices.end(); ++i)
     {
-        if (i->find("MYRIAD") != std::string::npos)
+        if (i->find(device) != std::string::npos)
         {
             const std::string name = ie.GetMetric(*i, METRIC_KEY(FULL_DEVICE_NAME)).as<std::string>();
             CV_LOG_INFO(NULL, "Myriad device: " << name);
-            return name.find("MyriadX") != std::string::npos  || name.find("Myriad X") != std::string::npos;
+
+            if (device == "MYRIAD")
+                return name.find("MyriadX") != std::string::npos  || name.find("Myriad X") != std::string::npos;
+            else if (device == "HDDL")
+                return name.find("HDDL") != std::string::npos;
```

Comment:
What is a full name for MyriadX based HDDL? I mean that there is also FPGA HDDL so we can distinguish them

Reply:
It is only 'HDDL'.
```
$ python3 /opt/intel/openvino/deployment_tools/inference_engine/samples/python/hello_query_device/hello_query_device.py
.
.
.
	Device: HDDL
	Metrics:
		AVAILABLE_DEVICES: 
		SUPPORTED_METRICS: AVAILABLE_DEVICES, SUPPORTED_METRICS, SUPPORTED_CONFIG_KEYS, FULL_DEVICE_NAME, OPTIMIZATION_CAPABILITIES, RANGE_FOR_ASYNC_INFER_REQUESTS, VPU_HDDL_DEVICE_NUM, VPU_HDDL_DEVICE_NAME, VPU_HDDL_DEVICE_THERMAL, VPU_HDDL_DEVICE_ID, VPU_HDDL_DEVICE_SUBCLASS, VPU_HDDL_DEVICE_MEMORY_TOTAL, VPU_HDDL_DEVICE_MEMORY_USED, VPU_HDDL_DEVICE_UTILIZATION, VPU_HDDL_DEVICE_TAG, VPU_HDDL_STREAM_ID
		SUPPORTED_CONFIG_KEYS: PERF_COUNT, VPU_INPUT_NORM, VPU_INPUT_BIAS, VPU_HDDL_GRAPH_TAG, VPU_HDDL_STREAM_ID, VPU_HDDL_DEVICE_TAG, VPU_HDDL_BIND_DEVICE, VPU_HDDL_RUNTIME_PRIORITY, VPU_HDDL_USE_SGAD, VPU_COPY_OPTIMIZATION, LOG_LEVEL, VPU_IGNORE_UNKNOWN_LAYERS, VPU_HW_STAGES_OPTIMIZATION, VPU_NONE_LAYERS, VPU_NUMBER_OF_SHAVES, VPU_NUMBER_OF_CMX_SLICES, VPU_HW_INJECT_STAGES, VPU_HW_POOL_CONV_MERGE, VPU_HW_STAGES_OPTIMIZATION, EXCLUSIVE_ASYNC_REQUESTS
		FULL_DEVICE_NAME: Intel Vision Accelerator Design with Intel Movidius VPUs
		OPTIMIZATION_CAPABILITIES: FP16
[16:36:35.6871][2839]I[ClientManager.cpp:159] client(id:2) registered: clientName=HDDLPluginForQuery socket=2
		RANGE_FOR_ASYNC_INFER_REQUESTS: 8, 150, 1
		VPU_HDDL_DEVICE_NUM: 2
		VPU_HDDL_DEVICE_NAME: 3.1, 3.2
		VPU_HDDL_DEVICE_THERMAL: 61.891929626464844, 62.51952362060547
		VPU_HDDL_DEVICE_ID: 224, 225
		VPU_HDDL_DEVICE_SUBCLASS: 0, 0
		VPU_HDDL_DEVICE_MEMORY_TOTAL: 483, 483
		VPU_HDDL_DEVICE_MEMORY_USED: 0, 0
		VPU_HDDL_DEVICE_UTILIZATION: 0.0, 0.0
		VPU_HDDL_DEVICE_TAG: , 
		VPU_HDDL_STREAM_ID: , 

	Default values for device configuration keys:
		PERF_COUNT: 
		VPU_INPUT_NORM: 
		VPU_INPUT_BIAS: 
		VPU_HDDL_GRAPH_TAG: 
		VPU_HDDL_STREAM_ID: 
		VPU_HDDL_DEVICE_TAG: 
		VPU_HDDL_BIND_DEVICE: 
		VPU_HDDL_RUNTIME_PRIORITY: 
		VPU_HDDL_USE_SGAD: 
		VPU_COPY_OPTIMIZATION: 
		LOG_LEVEL: 
		VPU_IGNORE_UNKNOWN_LAYERS: 
		VPU_HW_STAGES_OPTIMIZATION: 
		VPU_NONE_LAYERS: 
		VPU_NUMBER_OF_SHAVES: 
		VPU_NUMBER_OF_CMX_SLICES: 
		VPU_HW_INJECT_STAGES: 
		VPU_HW_POOL_CONV_MERGE: 
		VPU_HW_STAGES_OPTIMIZATION: 
		EXCLUSIVE_ASYNC_REQUESTS: 
```
When I insert the NCS2, I get
```
$ python3 /opt/intel/openvino/deployment_tools/inference_engine/samples/python/hello_query_device/hello_query_device.py
.
.
.
	Device: MYRIAD
	Metrics:
		RANGE_FOR_ASYNC_INFER_REQUESTS: 3, 6, 1
		OPTIMIZATION_CAPABILITIES: FP16
		DEVICE_THERMAL: UNSUPPORTED TYPE
		SUPPORTED_CONFIG_KEYS: PERF_COUNT, EXCLUSIVE_ASYNC_REQUESTS, DEVICE_ID, VPU_MYRIAD_PLATFORM, VPU_IGNORE_IR_STATISTIC, VPU_CUSTOM_LAYERS, CONFIG_FILE, VPU_MYRIAD_FORCE_RESET, VPU_PRINT_RECEIVE_TENSOR_TIME, LOG_LEVEL, VPU_HW_STAGES_OPTIMIZATION
		SUPPORTED_METRICS: RANGE_FOR_ASYNC_INFER_REQUESTS, OPTIMIZATION_CAPABILITIES, DEVICE_THERMAL, SUPPORTED_CONFIG_KEYS, SUPPORTED_METRICS, FULL_DEVICE_NAME, AVAILABLE_DEVICES
		FULL_DEVICE_NAME: Intel Movidius Myriad X VPU
		AVAILABLE_DEVICES: 1.1.2-ma2480

	Default values for device configuration keys:
		PERF_COUNT: OFF
		EXCLUSIVE_ASYNC_REQUESTS: OFF
		DEVICE_ID: 
		VPU_MYRIAD_PLATFORM: 
		VPU_IGNORE_IR_STATISTIC: OFF
		VPU_CUSTOM_LAYERS: 
		CONFIG_FILE: 
		VPU_MYRIAD_FORCE_RESET: OFF
		VPU_PRINT_RECEIVE_TENSOR_TIME: OFF
		LOG_LEVEL: LOG_NONE
		VPU_HW_STAGES_OPTIMIZATION: ON
```

Reply:
Got it. So, do you mind if `detectMyriadX_` will just return true in this way?
```cpp
return name.find("MyriadX") != std::string::npos  ||
       name.find("Myriad X") != std::string::npos ||
       name.find("HDDL") != std::string::npos;
```
because we use it only as 
```cpp
bool isMyriadX_ = detectMyriadX_("MYRIAD");
bool isHDDL_ = detectMyriadX_("HDDL");
if (isMyriadX_ || isHDDL_)
```

Reply:
Sure, sounds perfect. But why do you think this would resolve the failed builds?

Reply:
I have not checked failed builds - it just a part of code review. Let me see

Reply:
please apply the changes and let's see how ABI tests will go